dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -212,3 +213,82 @@ std::wstring Microsoft::Windows::AppNotifications::Helpers::RegisterComActivator
     THROW_IF_WIN32_ERROR(status);
     return registeredGuid;
 }
+
+wil::unique_cotaskmem_string Microsoft::Windows::AppNotifications::Helpers::ConvertUtf8StringToWideString(unsigned long length, const byte* utf8String)
+{
+    int size{ MultiByteToWideChar(
+        CP_UTF8,
+        0,
+        reinterpret_cast<PCSTR>(utf8String),
+        length,
+        nullptr,
+        0) };
+    THROW_LAST_ERROR_IF(size == 0);
+
+    wil::unique_cotaskmem_string wideString{ wil::make_unique_string<wil::unique_cotaskmem_string>(nullptr, size) };
+
+    size = MultiByteToWideChar(
+        CP_UTF8,
+        0,
+        reinterpret_cast<PCSTR>(utf8String),
+        length,
+        wideString.get(),
+        size);
+    THROW_LAST_ERROR_IF(size == 0);
+
+    return wideString;
+}
+
+winrt::Microsoft::Windows::AppNotifications::AppNotification Microsoft::Windows::AppNotifications::Helpers::ToastNotificationFromToastProperties(ABI::Microsoft::Internal::ToastNotifications::INotificationProperties* properties)
+{
+    unsigned int payloadSize{};
+    wil::unique_cotaskmem_array_ptr<byte> payload{};
+    THROW_IF_FAILED(properties->get_Payload(&payloadSize, &payload));
+
+    auto wide{ ConvertUtf8StringToWideString(payloadSize, payload.get()) };
+    winrt::hstring xmlPayload{ wide.get() };
+
+    winrt::Windows::Data::Xml::Dom::XmlDocument xmlDocument{};
+    xmlDocument.LoadXml(xmlPayload);
+
+    winrt::Microsoft::Windows::AppNotifications::AppNotification notification(xmlDocument);
+
+    wil::unique_hstring tag{};
+    properties->get_Tag(&tag);
```

Comment:
THROW_IF_FAILED for APIs that return HRESULT please

Reply:
thanks for cathing this. not sure why I missed those